test/unit/seeds/consent.test.ts

```diff
@@ -0,0 +1,156 @@
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Ahan Gupta <ahangupta.96@gmail.com>
+
+ --------------
+ ******/
+
+import Config from '../../../config/knexfile'
+import * as Knex from 'knex'
+import knex from 'knex'
+
+describe('testing Consent table', (): void => {
+  let db: knex<unknown[]>
+
+  beforeAll(async (): Promise<void> => {
+    db = knex(Config.test)
+    await db.migrate.latest()
+    await db.seed.run()
+  })
+
+  afterAll(async (): Promise<void> => {
+    db.destroy()
+  })
+
+  it('should properly select all the entries in the Consent table', async (): Promise<void> => {
+    expect(db).toBeDefined()
+    const users: Knex.QueryBuilder[] = await db.from('Consent').select('*')
+    expect(users.length).toEqual(3)
+    expect(users[0]).toMatchObject({
+      id: '123',
+      initiatorId: 'PISPA',
+      participantId: 'DFSPA',
+      credentialId: null,
+      credentialType: null,
+      credentialStatus: null,
+      credentialPayload: null,
+      credentialChallenge: null
+    })
+    expect(users[1]).toMatchObject({
+      id: '124',
+      initiatorId: 'PISPB',
+      participantId: 'DFSPA',
+      credentialId: '9876',
+      credentialType: 'FIDO',
+      credentialStatus: 'PENDING',
+      credentialPayload: null,
+      credentialChallenge: 'string_representing_challenge_a'
+    })
+    expect(users[2]).toMatchObject({
+      id: '125',
+      initiatorId: 'PISPC',
+      participantId: 'DFSPA',
+      credentialId: '9875',
+      credentialType: 'FIDO',
+      credentialStatus: 'VERIFIED',
+      credentialPayload: 'string_representing_public_key_a',
+      credentialChallenge: 'string_representing_challenge_b'
+    })
+  })
+})
+
+describe('testing that constraints are enforced in the consent table', async (): Promise<void> => {
+  let db: knex<unknown[]>
+
+  beforeAll(async (): Promise<void> => {
+    db = knex(Config.test)
+    await db.migrate.latest()
+    await db.seed.run()
+  })
+
+  afterAll(async (): Promise<void> => {
+    db.destroy()
+  })
+
+  it('should properly enforce the primary key constraint in the Consent table', async (): Promise<void> => {
+    expect(db).toBeDefined()
+    /* Tests for duplication */
+    await expect(db.from('Consent').insert({
+      id: '123',
+      initiatorId: 'PISPA',
+      participantId: 'DFSPA',
+      credentialId: null,
+      credentialType: null,
+      credentialStatus: null,
+      credentialPayload: null,
+      credentialChallenge: null
+    })).rejects.toMatchObject({
+      code: 'SQLITE_CONSTRAINT',
```

Comment:
Is this specific to SQLite? Is there a way to make sure that this is generic enough if someone changes the database engine/dialect?